src/styles/index.scss

```diff
@@ -1,3 +1,63 @@
 body {
+  display: flex;
   margin: 0;
+  align-items: center;
+  justify-content: center;
+  height: 100vh;
+}
+
+$day-size: 100px;
+$border-size: 1px;
+$padding: 10px;
+$hover-color: #ffbfcb;
+$day-color: #eee;
+$hover-transform: -20px;
+$transition-duration: 0.5s;
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: calc(7 * #{$day-size} + 7px);
+  padding: $padding;
+
+  .calendar__day {
+    box-sizing: border-box;
+    width: $day-size;
+    height: $day-size;
+    background-color: $day-color;
+    border: $border-size solid black;
+    position: relative;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    transition: all $transition-duration;
+
+    &:hover {
+      cursor: pointer;
+      background-color: $hover-color;
+      transform: translateY($hover-transform);
+    }
+  }
+
+  @for $i from 1 through 31 {
+    .calendar__day:nth-child(#{$i})::before {
+      content: '#{$i}';
+      font-family: Arial, sans-serif;
+      font-size: 30px;
+      margin-left: 1px;
+    }
+  }
+}
+
+@each $day, $index in (mon: 0, tue: 1, wed: 2, thu: 3, fri: 4, sat: 5, sun: 6) {
+  .calendar--start-day-#{$day} .calendar__day:nth-child(1) {
+    margin-left: calc(#{$day-size} * $index + 1px * $index);
```

Comment:
The margin-left calculation for the first day of the week is correctly using SCSS interpolation and variables. Ensure that the resulting CSS aligns the days correctly based on the starting day.